maven-model-builder/src/main/java/org/apache/maven/model/building/DefaultModelBuilder.java

```diff
@@ -1830,6 +1845,15 @@ private org.apache.maven.api.model.DependencyManagement doLoadDependencyManageme
                 return null;
             }
 
+            if (importSource instanceof FileModelSource && request.getRootDirectory() != null) {
+                URI sourceUri = ((FileModelSource) importSource).getLocationURI();
+                if (request.getRootDirectory().toUri().relativize(sourceUri) != sourceUri) {
```

Comment:
Fwiw, this test is unintuitive... a comment would be welcomed.  I had to check the javadoc...

Reply:
Fixed

Reply:
I don't want to be pushy, but what about the following ?
```
Path sourcePath = ((FileModelSource) importSource).getPath();
if (sourcePath.startsWith(request.getRootDirectory()) {
```

Reply:
Pushed, so who's pushy now?